metrics/ci_integrations/test/client/github_actions/github_actions_client_test.dart

```diff
@@ -309,6 +309,124 @@ void main() {
       },
     );
 
+    // start
+    test(
+      ".fetchWorkflowRunsWithConclusion() fails if there is no workflow with such identifier",
+      () async {
+        final interactionResult = await client.fetchWorkflowRunsWithConclusion(
+          "test",
+        );
+        final isError = interactionResult.isError;
+
+        expect(isError, isTrue);
+      },
+    );
+
+    test(
+      ".fetchWorkflowRunsWithConclusion() applies the default per page parameter if it is not specified",
+      () async {
+        final interactionResult = await client.fetchWorkflowRunsWithConclusion(
+          workflowId,
+        );
+        final runsPage = interactionResult.result;
+
+        expect(runsPage.perPage, isNotNull);
+      },
+    );
+
+    test(
+      ".fetchWorkflowRunsWithConclusion() returns a workflow runs page with the given per page parameter",
+      () async {
+        const expectedPerPage = 3;
+
+        final interactionResult = await client.fetchWorkflowRunsWithConclusion(
+          workflowId,
+          perPage: expectedPerPage,
+        );
+        final runsPage = interactionResult.result;
+
+        expect(runsPage.perPage, equals(expectedPerPage));
+      },
+    );
+
+    test(
+      ".fetchWorkflowRunsWithConclusion() fetches the first page if the given page parameter is null",
+      () async {
+        final interactionResult = await client.fetchWorkflowRunsWithConclusion(
+          workflowId,
+          page: null,
+        );
+        final runsPage = interactionResult.result;
+
+        expect(runsPage.page, equals(1));
+      },
+    );
+
+    test(
+      ".fetchWorkflowRunsWithConclusion() fetches the first page if the given page parameter is less than zero",
```

Comment:
Should we test that this method fetches the first page if the given `page` parameter equals `0`?